src/event_imp.rs

```diff
@@ -1044,6 +1044,216 @@ fn test_debug_ready() {
     assert_eq!("Writable", format!("{:?}", Ready::writable()));
 }
 
+
+/// Interests used in registering.
+///
+/// Interests are used in registering [`Evented`] handles with [`Poll`],
+/// they indicate what readiness should be monitored for. For example if a
+/// socket is registered with readable interests and the socket becomes
+/// writable, no event will be returned from [`poll`].
+///
+/// [`Poll`]: struct.Poll.html                                                 
+/// [`readable`]: #method.readable
+/// [`writable`]: #method.writable
+#[derive(Copy, PartialEq, Eq, Clone, PartialOrd, Ord)]
+pub struct Interests(usize);
```

Comment:
In mio-st I've made this `NonZeroU8` to make `Option<Interests>` the same size as `Interests`. But that can come later (in a future pr).

Reply:
I will do that in a future pr.